api/v1alpha4/clusterclass_webhook.go

```diff
@@ -173,7 +300,16 @@ func (in *ClusterClass) validate(old *ClusterClass) error {
 		oldClassNames.Insert(oldClass.Class)
 	}
 
-	// Makes sure no additional changes were applied.
+	return allErrs
+}
+
+func (in ClusterClass) validateAdditionalChanges(old *ClusterClass) field.ErrorList {
```

Comment:
This name kind of confuses me (additional change, seems like a catch all..) but I don't have better suggestions...

Reply:
How about `validateCompatibleSpecChanges`?.  This function could be extended/modified when we introduce the concept of compatibility among cluster classes.

Reply:
this works for me!

Reply:
I'm not sure if that not might lead to an association to all our compatibility rules in the proposal. WDYT about `validateImmutableFields` ?

Reply:
I am fine with either, however since the checks we have here can be considered as compatibility change (immutable == all changes are incompatible) and as we are introducing the concept of compatibility across clusterclasses we can modify this functions to introduce the new changes gradually.  wdyt?

Reply:
How about having `validatecompatibleSpecChanges` call `validateImmutableFields`? I believe even in compatibility check we will have to make sure some fields are immutable and the corresponding checked can to under `validateImmutableFields`?

Reply:
Hm I'm not sure if multiple layers here will help :). Actually according to the current state of the compatibility PR only parts of the infrastructure and controlplane ref will be immutable. Hm I think the best idea is probably really to call it `validateCompatibleSpecChanges`.

In this case `validateMachineDeployments` (at least conceptually) is also a part of our compatibility definition, so we maybe should move the call of this func into `validateCompatibleSpecChanges`. 

So in this case the `validate` func would have the following calls:
* call funcs to validate if the fields themselves are valid 
* call `validateCompatibleSpecChanges` to check if the changed fields are compatible (this should only be necessary during update)

Reply:
Sounds good! 👍🏼